packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -70,12 +73,30 @@ export const gcTestModeKey = "Fluid.GarbageCollection.GCTestMode";
 const writeAtRootKey = "Fluid.GarbageCollection.WriteDataAtRoot";
 // Feature gate key to expire a session after a set period of time.
 export const runSessionExpiryKey = "Fluid.GarbageCollection.RunSessionExpiry";
-// Feature gate key to disable expiring session after a set period of time, even if expiry value is present
+// Feature gate key to disable expiring session after a set period of time, even if expiry value is present.
 export const disableSessionExpiryKey = "Fluid.GarbageCollection.DisableSessionExpiry";
 // Feature gate key to write the gc blob as a handle if the data is the same.
 export const trackGCStateKey = "Fluid.GarbageCollection.TrackGCState";
 // Feature gate key to turn GC sweep log off.
-const disableSweepLogKey = "Fluid.GarbageCollection.DisableSweepLog";
+export const disableSweepLogKey = "Fluid.GarbageCollection.DisableSweepLog";
+/**
+ * Feature gate key to enable closing the container if SweepReady objects are used.
+ * Only known values are accepted, otherwise return undefined.
+ *
+ * mainContainer: Detect these errors only in the main container
+ */
+ export const sweepReadyUsageDetectionSetting = {
+    read(config: IConfigProvider): "mainContainer" | undefined {
+        const sweepReadyUsageDetectionKey = "Fluid.GarbageCollection.Dogfood.SweepReadyUsageDetection";
+        const value = config.getString(sweepReadyUsageDetectionKey);
+        switch (value) {
+            case "mainContainer":
```

Comment:
It's probably better to use an enum for this.

Reply:
I'm actually gonna switch it to checking the value for substring to be easier to use. And this fn will spit out an object like { mainContainer: true, summarizer: false, }

Reply:
Also, what are the values going to be once you add summarizer? I think we should have 2 modes:
1. Summarizer and non-summarizer clients. Maybe the keyword to use here is "all" or "everywhere" or "allClients"?
2. Only summarizer clients - "summarizer" or "summarizerClient".

Reply:
I'm curious as to why this feature flag is more complicated and what advantages it has over the original system.

Reply:
I could also just have independent boolean flags for main v. summarizer.

Reply:
I think one flag is better with different values. If you use objects you can run into issues in ECS. You will have to JSON.stringify them in ECS and then parse it here last time I checked.

Reply:
Ok.  Then I'll stick with the plan I said above.  See https://github.com/markfields/FluidFramework/blob/5af5b089a87711b06a70934ff78e82abd182d28b/packages/runtime/container-runtime/src/gcSweepReadyUsageDetection.ts#L24-L35

Reply:
Oh this is what you meant. I thought you meant adding an object directly in ECS :-).
With the code you shared, either one of these will be set right? How would you set both - summarizerAndmainContainer?